pkg/patch/datadog.go

```diff
@@ -0,0 +1,123 @@
+package patch
+
+import (
+	"bufio"
+	"bytes"
+	"text/template"
+)
+
+// Creating a template to avoid relying on an extra ConfigMap
+const datadogTemplate = `
+tracing:
+  http:
+    name: envoy.tracers.datadog
+    config:
+      collector_cluster: datadog_agent
+      service_name: envoy
+static_resources:
+  clusters:
+  - name: datadog_agent
+    connect_timeout: 1s
+    type: strict_dns
+    lb_policy: round_robin
+    load_assignment:
+      cluster_name: datadog
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+           address:
+            socket_address:
+             address: {{ .Address }}
+             port_value: {{ .Port }}        
+`
+
+const injectDatadogTemplate = `
+{
+  "command": [
+    "sh",
+    "-c",
+    "cat <<EOF >> /tmp/envoy/envoyconf.yaml{{ .Config }}EOF\n\ncat /tmp/envoy/envoyconf.yaml\n"
```

Comment:
Can you please change the config file name? if both tracers are enabled, the init containers will overwrite each others config.

Reply:
I can, but from my understanding Envoy only supports 1 tracer at a time ?
Please correct me if I am wrong, if there is indeed a way, I could make the `append` more generic so that both tracers and clusters can be listed.
Otherwise, I can duplicate the logic to add a specific file, but it will still bottleneck as we reference two different files with the same env var `ENVOY_STATS_CONFIG_FILE`

Reply:
Ah you are right! Ok we need to add to the docs that users have to chose a single tracer. In the Inject Helm chart this restriction will be easy to implement.

Reply:
Okay, I can do that yes

Reply:
It should be enforced in more than just the helm chart, though.  Let's add a check to error if both are enabled.